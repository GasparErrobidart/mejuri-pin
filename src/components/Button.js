import styled     from 'styled-components';

const StyledButton = styled.div`
  transition: all 300ms ease-in-out;
  font-size: 1rem;
  font-weight: 300;
  padding: 5px 30px;
  margin: 10px;
  border: 1px solid ${({theme})=> theme.colors.middleGray };
  background: transparent;
  &:hover, &:active{
    color: ${({theme})=> theme.colors.darkGray };
    border: 1px solid ${({theme})=> theme.colors.darkGray };
    background: white;
  }
`

export default StyledButton;
