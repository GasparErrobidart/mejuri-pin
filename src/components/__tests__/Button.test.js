import '@testing-library/jest-dom'
import 'jest-styled-components'

import React                 from 'react'
import {ThemeProvider}       from 'styled-components'
import { cleanup }           from '@testing-library/react'
import renderer              from 'react-test-renderer'

import theme                 from '../GlobalStyle/themes/mejuri-pin'
import Button                from '../Button'


afterEach(cleanup)

test('Renders "Button" component', () => {
  const testText        = "Button Test Text"
  const component       = renderer.create(<ThemeProvider theme={theme}><Button>{testText}</Button></ThemeProvider>);
  const tree            = component.toJSON()
  expect(tree).toMatchSnapshot();
});
