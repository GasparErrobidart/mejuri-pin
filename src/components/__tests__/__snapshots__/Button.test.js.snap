// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "Button" component 1`] = `
.c0 {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  font-size: 1rem;
  font-weight: 300;
  padding: 10px 10px;
  margin: 10px;
  color: #303030;
  box-shadow: 0px 2px 4px #aaa;
  text-transform: uppercase;
  background: transparent;
  cursor: pointer;
}

.c0:hover,
.c0:active,
.c0.active {
  box-shadow: 0px 0px 2px #aaa;
  color: #888888;
  background: #F3F3F3;
}

@media (min-width:768px) {
  .c0 {
    padding: 10px 30px;
  }
}

<div
  className="c0"
>
  Button Test Text
</div>
`;
